compiler/lexer.nim

```diff
@@ -882,15 +882,61 @@ proc endOperator(L: var Lexer, tok: var Token, pos: int,
   else: tok.tokType = TokType(tok.ident.id - oprLow + ord(tkColon))
   L.bufpos = pos
 
+const
+  UnicodeOperatorStartChars = {'\226', '\194', '\195'}
+    # the allowed unicode characters ("∙ ∘ × ★ ⊗ ⊘ ⊙ ⊛ ⊠ ⊡ ∩ ∧ ⊓ ± ⊕ ⊖ ⊞ ⊟ ∪ ∨ ⊔")
+    # all start with one of these.
+
+proc unicodeOprLen(L: var Lexer; tok: var Token; pos: int): int =
+  if unicodeOperators notin L.config.features: return 0
+  result = 0
+  case L.buf[pos]
+  of '\226':
+    if L.buf[pos+1] == '\136':
+      if L.buf[pos+2] == '\152': result = 3 # ∘
+      elif L.buf[pos+2] == '\153': result = 3 # ∙
+      elif L.buf[pos+2] == '\167': result = 3 # ∧
+      elif L.buf[pos+2] == '\168': result = 3 # ∨
+      elif L.buf[pos+2] == '\169': result = 3 # ∩
+      elif L.buf[pos+2] == '\170': result = 3 # ∪
+    elif L.buf[pos+1] == '\138':
+      if L.buf[pos+2] == '\147': result = 3 # ⊓
+      elif L.buf[pos+2] == '\148': result = 3 # ⊔
+      elif L.buf[pos+2] == '\149': result = 3 # ⊕
+      elif L.buf[pos+2] == '\150': result = 3 # ⊖
+      elif L.buf[pos+2] == '\151': result = 3 # ⊗
+      elif L.buf[pos+2] == '\152': result = 3 # ⊘
+      elif L.buf[pos+2] == '\153': result = 3 # ⊙
+      elif L.buf[pos+2] == '\155': result = 3 # ⊛
+      elif L.buf[pos+2] == '\158': result = 3 # ⊞
+      elif L.buf[pos+2] == '\159': result = 3 # ⊟
+      elif L.buf[pos+2] == '\160': result = 3 # ⊠
+      elif L.buf[pos+2] == '\161': result = 3 # ⊡
+    elif L.buf[pos+1] == '\152' and L.buf[pos+2] == '\133': result = 3 # ★
+  of '\194':
+    if L.buf[pos+1] == '\177': result = 2 # ±
+  of '\195':
+    if L.buf[pos+1] == '\151': result = 2 # ×
```

Comment:
I would leave out the ones of len `2` and simplify this whole proc. 🤔

Reply:
I followed the spec as outlined in the RFC.

Reply:
What, those two are two of the most important new symbols :confused:.